### PR TITLE
[ESIMD] Fix lsc_slm_gather.cpp test that failed with opencl:gpu

### PR DESCRIPTION
**Pull request by v-klochkov**

### CODE REVIEW

File: sycl/test-e2e/ESIMD/lsc/Inputs/lsc_slm_load.hpp [hunk 1]
```diff
@@ -146,7 +146,7 @@ bool test(queue Q, uint32_t PMask = ~0) {
           Tuint ExpectedVal =
               IsMaskSet ? (ExpectedValBase + ChannelId + J * NChannels) & VMask
                         : sycl::bit_cast<Tuint>(MergeValue);
-          Tuint ComputedVal = sycl::bit_cast<Tuint>(Out[OutIndex]);
+          Tuint ComputedVal = sycl::bit_cast<Tuint>(Out[OutIndex]) & VMask;
```
Comment (sarnex) [inline sycl/test-e2e/ESIMD/lsc/Inputs/lsc_slm_load.hpp:149]: Is this related to what we discussed with Victor recently?

Comment (v-klochkov) [reply sycl/test-e2e/ESIMD/lsc/Inputs/lsc_slm_load.hpp:149]: I don't remember which test was shown to Victor.
This test failed on PVC with opencl and I fixed the issue.

Comment (v-klochkov) [reply sycl/test-e2e/ESIMD/lsc/Inputs/lsc_slm_load.hpp:149]: If this one is different case, then it is similar.

**Comment by v-klochkov** [comment]
Fail in AMD/Hip CI run is unrelated to the fix in ESIMD test done here.

**Review by sarnex — APPROVED** [review]

**Review by turinevgeny — APPROVED** [review]

File: sycl/test-e2e/ESIMD/lsc/Inputs/lsc_slm_load.hpp (hunk 1 above)
Comment (v-klochkov) [reply sycl/test-e2e/ESIMD/lsc/Inputs/lsc_slm_load.hpp:149]: Thank you.
BTW, @turinevgeny says is the same test that we looked at yesterday.

**Pull request merged**